packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -109,10 +110,12 @@ export function useAsyncData<
   options.lazy = options.lazy ?? false
   options.immediate = options.immediate ?? true
 
+  options.forcePayloadExtraction = options.forcePayloadExtraction ?? false
+
   // Setup nuxt instance payload
   const nuxt = useNuxtApp()
 
-  const getCachedData = () => nuxt.isHydrating ? nuxt.payload.data[key] : nuxt.static.data[key]
+  const getCachedData = () => (nuxt.isHydrating || options.forcePayloadExtraction) ? nuxt.payload.data[key] : nuxt.static.data[key]
```

Comment:
I'm reluctant to merge this PR as it's really a feature to work around a bug.

What about instead supporting a `cache`/`store` option that defaults to `nuxt.isHydrating ? nuxt.payload.data : nuxt.static.data`? Then this could be set to a custom location, or forced to `nuxt.payload.data`, which would have the same effect as forcing payload extraction.

It could be part of https://github.com/nuxt/nuxt/issues/15445.

Thoughts @pi0?

Reply:
More generic `cache` option with better defaults sound reasonably better alternative approach.